### ISSUE TITLE
Payload changes when menu is re-rendered.

### ISSUE THREAD

**Issue by adithyagenie** [grammyjs/menu#39]
Any dynamic menu running with the help of payloads such as this:

```js
function sampleMenu() {
  return new Menu("allist") 
  .dynamic(async (ctx) => {
    const range = new MenuRange();
    const payload = ctx.match.toString();
    console.log(payload);
    range.text({text:"Button 1", payload: `${payload}_2`});
    return range;
  });
}
```

When this button is pressed and it results in an menu outdated scenario, the payload of it should ideally be what it was initially generated with (say initially the payload was 10). But what happens is the same menu is re-generated with the new payload (that is 10_2 in this case). This breaks several stuff in generating the menu.

**Comment by KnorpelSenf** [comment]
Damn, you're right. Can you fix it? :)

**Comment by adithyagenie** [comment]
> Damn, you're right. Can you fix it? :)

I have currently just used a temporary session variable which stores the payload when it initially generated and forced it to generate with the correct payload by using a custom `onMenuOutdated` function.

**Comment by KnorpelSenf** [comment]
Hang on, I think I initially misread your issue. The old payload is gone and there is no way to recover it. The menu always operates on up-to-date data, and we do not store a history of previous input values or let users roll back to earlier renders of the menu.

If you do need to access outdated values such as old payloads, then it is indeed required to store data in a database, which isn't something the menu plugin concerns itself with.

> This breaks several stuff in generating the menu.

Can you elaborate why your current payload is incorrect?

**Comment by adithyagenie** [comment]
So to replicate the exact scenario, this is a minified version of my pagination menu code -

```
To replicate the exact same scenario I was talking about, 

I had two functions namely

function call() {
    return new Menu("call").dynamic((ctx, range) => {
        range.submenu({text: "go", payload: "1"}, "yo3", (ctx1) => ctx1.editMessageText("ran"));
    });
}

function call2() {
    return new Menu("callsub")
        .dynamic(ctx => {
            console.log(`Payload in callsub:: ${ctx.match.toString()}`);
            return new MenuRange<MyContext>()
                .submenu({text: ctx.match.toString(), payload: `${ctx.match.toString()}_1`}, "yo", (ctx) => ctx.);
        });
}
```

Here, "call" is the main menu which calls the dynamic submenu "callsub". 
**callsub** generates buttons based on the payload it recieves (this is what I did in my case of a pagination bot, it generates page buttons based on current page which is received by it in payload). 

**Expected behaviour:** Upon pressing the button which I made it to be the payload (page number in my case), it should be navigating because the menu is the same for the page it was in before pressing the button as it was when it initially rendered.

But instead, when pressing the "1" button generated by "callsub", it says menu outdated because **it re-renders the same menu with payload of the button which is 1_1**, which changes the button text, hence rendering the menu as "outdated". 
What should be happening instead is that, **it should be re-rendered for checking with "1" as a payload instead of the new payload** which was for the button "1_1".

Hope this makes it a bit more clearer.
